packages/ui/src/i18n/init.ts

```diff
@@ -18,7 +18,18 @@ const initI18n = async (languageSettings?: LanguageInfo) => {
       ? { ...baseOptions, lng: languageSettings.fixedLanguage }
       : baseOptions;
 
-  return i18next.use(initReactI18next).use(LanguageDetector).init(options);
+  const i18n = i18next.use(initReactI18next).use(LanguageDetector).init(options);
+
+  // @ts-expect-error - i18next doesn't have a type definition for this. called after i18next is initialized
+  i18next.services.formatter.add('zhOrSpaces', (value: string, lng) => {
```

Comment:
It doesn't have to be this cumbersome when dealing with spaces. You can build your string with as many spaces as you want, and browser will automatically ignore the excess whitespace and only render one. For example:
```
<a>Sign in with     mobile       or          email</a>
```
Then in runtime, you will find your text link on the browser is still
```
Sign in with mobile or email
```

On the contrary, if you want to keep those whitespaces instead of collapsing them, you should use `<pre>` or use css style `whitespace: pre`.

Reply:
Well, after discussion I realized there are cases that we have more than 2 methods, and we don't want to have extra whitespace around comma (e.g. email , phone number). So let's stick to the current implementation for now